### PR TITLE
Coverity fixes for "Copy into fixed sized buffer" in uri_sig and jtest

### PR DESCRIPTION
**Pull request by PSUdaemon**

### CODE REVIEW

File: plugins/experimental/url_sig/url_sig.c [hunk 1]
```diff
@@ -16,6 +16,13 @@
   limitations under the License.
  */
 
+#define min(a, b)           \
```
Comment (zwoop) [inline plugins/experimental/url_sig/url_sig.c]: Clever, but at this point, wouldn't it be easier / better to just make an inline min() function?

File: plugins/experimental/url_sig/url_sig.c [hunk 2]
```diff
@@ -264,7 +271,7 @@ getAppQueryString(char *query_string, int query_length)
     return NULL;
```
Comment (jpeach) [inline plugins/experimental/url_sig/url_sig.c]: The test above should be:
```C
if (query_length > (sizeof(buf) - 1)) {
  ...
}
```

File: plugins/experimental/url_sig/url_sig.c [hunk 3]
```diff
@@ -497,7 +504,7 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
 
   signed_part[strlen(signed_part) - 1] = '?'; // chop off the last /, replace with '?'
```
Comment (jpeach) [inline plugins/experimental/url_sig/url_sig.c]: `signed_part` might not be nul-terminated now, if we were unlucky enough to exactly fill the buffer.

File: tools/jtest/jtest.cc [hunk 4]
```diff
@@ -2118,7 +2118,7 @@ extract_urls(char *buf, int buflen, char *base_url)
         char *he = strchr(base_url + 8, '/');
         if (!memchr(base_url, '.', he - base_url)) {
           char t[512];
-          strcpy(t, base_url);
+          strncpy(t, base_url, sizeof(t) - 1);
```
Comment (jpeach) [inline tools/jtest/jtest.cc]: `t` might not be nul-terminated.

File: tools/jtest/jtest.cc [hunk 5]
```diff
@@ -3296,7 +3296,7 @@ make_url_client(const char *url, const char *base_url, bool seen, bool unthrottl
       curl[strlen(curl) - 1] = 0;
     }
   } else {
-    strcpy(curl, url);
+    strncpy(curl, url, sizeof(curl) - 1);
```
Comment (jpeach) [inline tools/jtest/jtest.cc]: `curl` might not be nul-terminated.

File: tools/jtest/jtest.cc (hunk 4 above)
Comment (PSUdaemon) [inline tools/jtest/jtest.cc]: So `t` is the destination. Additionally, if `base_url` was not NULL terminated before then this changes nothing. What this fix does do is not overrun `t`.

File: tools/jtest/jtest.cc (hunk 5 above)
Comment (PSUdaemon) [inline tools/jtest/jtest.cc]: Same as above.

**Comment by PSUdaemon** [comment]
@jpeach I agree, `url_sig.c` could stand a lot of work, but I'm trying not to pull that thread in this PR.

File: plugins/experimental/url_sig/url_sig.c (hunk 2 above)
Comment (PSUdaemon) [inline plugins/experimental/url_sig/url_sig.c]: Agree. I can fix that since it effected the `strncpy` below.

File: tools/jtest/jtest.cc (hunk 4 above)
Comment (jpeach) [inline tools/jtest/jtest.cc]: I meant that if  `strlen(base_url) >= 511`, then `t` won't be terminated.

**Comment by PSUdaemon** [comment]
[approve ci]

File: tools/jtest/jtest.cc (hunk 4 above)
Comment (PSUdaemon) [inline tools/jtest/jtest.cc]: Ah, I see. Yeah, no worse than before, but I think I can fix it.

**Pull request merged**